lib/_stream_wrap.js

```diff
@@ -13,57 +13,44 @@ const errors = require('internal/errors');
 
 function StreamWrap(stream) {
   const handle = new JSStream();
-
-  this.stream = stream;
-
-  this._list = null;
-
-  const self = this;
-  handle.close = function(cb) {
-    debug('close');
-    self.doClose(cb);
-  };
-  handle.isAlive = function() {
-    return self.isAlive();
-  };
-  handle.isClosing = function() {
-    return self.isClosing();
-  };
-  handle.onreadstart = function() {
-    return self.readStart();
-  };
-  handle.onreadstop = function() {
-    return self.readStop();
-  };
-  handle.onshutdown = function(req) {
-    return self.doShutdown(req);
-  };
-  handle.onwrite = function(req, bufs) {
-    return self.doWrite(req, bufs);
-  };
-
-  this.stream.pause();
-  this.stream.on('error', function onerror(err) {
-    self.emit('error', err);
-  });
-  this.stream.on('data', function ondata(chunk) {
+  const onData = (chunk) => {
     if (!(chunk instanceof Buffer)) {
       // Make sure that no further `data` events will happen
-      this.pause();
-      this.removeListener('data', ondata);
+      stream.pause();
+      stream.removeListener('data', onData);
 
-      self.emit('error', new errors.Error('ERR_STREAM_HAS_STRINGDECODER'));
+      this.emit('error', new errors.Error('ERR_STREAM_HAS_STRINGDECODER'));
       return;
     }
 
     debug('data', chunk.length);
-    if (self._handle)
-      self._handle.readBuffer(chunk);
+    if (this._handle)
+      this._handle.readBuffer(chunk);
+  };
+
+  handle.close = (cb) => {
+    debug('close');
+    this.doClose(cb);
+  };
+  handle.isAlive = () => this.isAlive();
+  handle.isClosing = () => this.isClosing();
+  handle.onreadstart = () => this.readStart();
+  handle.onreadstop = () => this.readStop();
+  handle.onshutdown = (req) => this.doShutdown(req);
+  handle.onwrite = (req, bufs) => this.doWrite(req, bufs);
+
+  this._list = null;
+
+  this.stream = stream;
+  this.stream.pause();
+  this.stream.on('error', (err) => {
+    this.emit('error', err);
   });
-  this.stream.once('end', function onend() {
+  this.stream.on('data', onData);
+  this.stream.once('end', () => {
     debug('end');
-    if (self._handle)
-      self._handle.emitEOF();
+    if (this._handle)
+      this._handle.emitEOF();
   });
```

Comment:
I think this is way more debuggable as with the previous pattern, as it is a named function.